.vib/pinniped/ginkgo/integration_suite_test.go

```diff
@@ -0,0 +1,217 @@
+	kubeconfigContent, _ := os.ReadFile(*kubeconfig)
+	scriptContent, _ := os.ReadFile("./scripts/pinniped-auth.sh")
+
+	podData := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: *namespace,
+			Name:      "vib-minideb-" + fmt.Sprint(rand.Intn(100)),
+			Labels: map[string]string{
+				"app": "vib-minideb",
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:       "vib-minideb",
+					Image:      "docker.io/bitnami/minideb",
+					WorkingDir: "/tmp",
+					Command: []string{
+						"/bin/bash", "-c", "printenv PINNIPED_SCRIPT | base64 -d > pinniped-script.sh && chmod +x pinniped-script.sh && ./pinniped-script.sh && sleep infinity"},
+					Env: []v1.EnvVar{
+						{
+							Name:  "KUBECONFIG",
+							Value: fmt.Sprint(b64.StdEncoding.EncodeToString([]byte(kubeconfigContent))),
+						},
+						{
+							Name:  "PINNIPED_SCRIPT",
+							Value: fmt.Sprint(b64.StdEncoding.EncodeToString([]byte(scriptContent))),
+						},
+						{
+							Name:  "PINNIPED_USERNAME",
+							Value: *authUser,
+						},
+						{
+							Name:  "PINNIPED_PASSWORD",
+							Value: *authPassword,
+						},
+					},
+				},
+			},
+		},
+	}
+	result, err := c.Pods(*namespace).Create(ctx, podData, metav1.CreateOptions{})
+	if err != nil {
+		panic(fmt.Sprintf("There was an error creating the Pod: %s", err))
+	}
+	return result
+}
+
+func isPodRunning(ctx context.Context, c cv1.PodsGetter, podName string) (bool, error) {
+	pod, err := c.Pods(*namespace).Get(ctx, podName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("There was an error obtaining the Pod %q", podName)
+		return false, err
+	}
+	if pod.Status.Phase == "Running" {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func getPodsByLabelOrDie(ctx context.Context, c cv1.PodsGetter, selector string) v1.PodList {
+	output, err := c.Pods(*namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Printf("Obtained list of pods with label %q\n", selector)
+
+	return *output
+}
+
+func getContainerLogsOrDie(ctx context.Context, c cv1.PodsGetter, podName string, containerName string) []string {
+	var output []string
+	tailLines := int64(100)
+
+	readCloser, err := c.Pods(*namespace).GetLogs(podName, &v1.PodLogOptions{
+		Container: containerName,
+		Follow:    false,
+		TailLines: &tailLines,
+	}).Stream(ctx)
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Printf("Obtained %q pod's logs\n", podName)
+
+	defer readCloser.Close()
+
+	scanner := bufio.NewScanner(interruptableReader{ctx, readCloser})
+	for scanner.Scan() {
+		output = append(output, scanner.Text())
+	}
+	if scanner.Err() != nil {
+		panic(scanner.Err())
+	}
+	return output
+}
+
+func containerLogsContainPattern(ctx context.Context, c cv1.PodsGetter, podName string, pattern string) (bool, error) {
+	containerLogs := getContainerLogsOrDie(ctx, c, podName, "vib-minideb")
```

Comment:
The container from where to obtain the logs should be passed as an argument instead of being hardcoded, given that `podName` is and the name of the function is generic.